enhancements/network/service-egress-traffic-steering.md

```diff
@@ -60,14 +61,20 @@ The ingress part will be handled by a LoadBalancer provider, such as MetalLB, th
 Taking MetalLB as an example for a LoadBalancer provider, the user will need to create their `L2Advertisement` and/or `BGPAdvertisements` with the `nodeSelector` field pointing to that label. That way only the node holding the label will be used for announcing the LoadBalancer service ingress IP.
 It is worth noting that in MetalLB's case, a given LoadBalancer service can be announced by multiple L2 and BGP advertisements, possibly being (even accidently) announced from multiple nodes. For our use-case the user MUST take care of configuring their MetalLB resources in a way that the service is announced only by the node holding the label - a full example is detailed in [Usage Example](#Usage-Example).
 
-To achieve these goals, we introduce a new annotation for users to set on LoadBalancer services: `k8s.ovn.org/egress-service`, which can be either empty or contain a `nodeSelector` field: `'{"nodeSelector":{"matchLabels":{"color": "green"}}}'` that allows limiting the nodes that can be selected to handle the service's traffic.
-By specifying the `nodeSelector` field, only a node whose labels match the specified selectors can be selected for handling the service's traffic as explained earlier.
-By not specifying the `nodeSelector` field any node in the cluster can be chosen to manage the service's traffic.
+To achieve these goals, we introduce a new resource `EgressService` for users to create alongside LoadBalancer services which can be either empty or contain optional fields:
+- `nodeSelector`: allows limiting the nodes that can be selected to handle the service's traffic.
+When present only a node whose labels match the specified selectors can be selected for handling the service's traffic as explained earlier.
+When the field is not specified any node in the cluster can be chosen to manage the service's traffic.
+
+- `routingTable`: The routing table to use for the service.
```

Comment:
can we make this even more generic? like 

`network`: The network which this service should send egress and corresponding ingress replies to. This is typically implemented as VRF mapping, which may be represented as a numeric id or string name. By default the egress traffic is routed via the default network in the host.

Reply:
maybe I misunderstand - I don't mind too much, but won't it make the field more vague while still being the same thing (matching on a routing table)?

Reply:
yeah but a user does not need to be concerned with routing tables. They only need to know I have multiple networks attached to each NIC, and I want traffic to go out that network. How that is managed under the covers I think we should keep that separated from the API.

Reply:
Hmm but isn't the value itself referring directly to a host routing table by name or id and not referring to a NIC or network? So the user would be forced to understand what that is?
When I read routing table I also read policy based routing as a broader routing application than just network routing.
So referring to a network seems to play down the potential usability of this?
Or maybe I am misunderstanding how this is supposed to be used.

Reply:
Oh, but I am OK with the intention of the description you give in your original comment
I think this is a nice compromise of downplaying the potential usability (read complexity) with guiding the user on how to use it.
I would suggest this modification:
```
network: The network which this service should send egress and corresponding ingress replies to. This is typically implemented as VRF mapping, representing a numeric id or string name of a routing table. By omission, egress traffic uses the default host routing.
```

Reply:
Agreed that if the intention of the feature is to route traffic to different networks, and we chose to implement it with routing tables, that should stay as an implementation detail. In the future this could be implemented differently.

Reply:
@oribon can you please update PR with the text provided?